tests/test_precision.py

```diff
@@ -34,9 +34,11 @@ def run_and_measure_memory(precision: Precision) -> int:
     return torch.cuda.max_memory_allocated()
 
 
-@pytest.mark.timeout(60)
+@pytest.mark.timeout(90)
 @pytest.mark.gpu
-def test_fp16_mixed():
+def test_precision():
```

Comment:
we usually prefer to have each test only test/assert one condition, so could you use parameterize to run both precisions? e.g.
```
@pytest.mark.parametrize(precision, ['Precision.AMP', 'Precision.BF16'])
def test_precision_memory(precision):
     ...
```

This would run the test twice.

Side question: would test just fail on V100 GPUs that don't wupport bfloat16? we may want to find the CUDA compatibility number for the device and skip if bf16 not supported?

Reply:
I've updated the test to check the PyTorch version number. If it's less than 1.10, it skips the bfloat16 test. This way, the test should not fail when the user is running an old version of PyTorch.
Answer to side question: https://discuss.pytorch.org/t/bfloat16-native-support/117155. It looks like CUDA & PyTorch don't provide a way to see if the GPU hardware natively supports bfloat16. I believe that PyTorch > 1.10 will fallback when operations don't have native support, so it shouldn't fail. This can be tested.